### PR TITLE
refactor(relayer): Factor out common listener utils

### PR DESCRIPTION
**Pull request by pxrl**
This is a pre-emptive change made to make it easier to reuse common components in alternative listener implementations. Examples for a viem-based listener, as well as an eventual Solana listener.

There are subsequent changes in the pipeline to factor out evm-specific parts.

### CODE REVIEW

File: src/libexec/util.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Given the inputs for a SpokePoolClient update, consolidate the inputs into a message and submit it to the parent
+ * process (if defined).
+ * @param blockNumber Block number up to which the update applies.
+ * @param currentTime The SpokePool timestamp at blockNumber.
+ * @param events An array of Log objects to be submitted.
+ * @returns void
+ */
+export function postEvents(blockNumber: number, oldestTime: number, currentTime: number, events: Log[]): void {
+  if (!isDefined(process.send) || stop) {
```
Comment (pxrl) [inline src/libexec/util.ts:14]: `stop` isn't available here.

Comment (pxrl) [reply src/libexec/util.ts]: Fixed: https://github.com/across-protocol/relayer/pull/1841/commits/ff6f406daba69b2bddab0fdbb059957a3b4d5be8

File: src/libexec/util/evm/util.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  const fromBlock = Math.max(toBlock - (lookback ?? deploymentBlock), deploymentBlock);
+  assert(toBlock > fromBlock, `${toBlock} > ${fromBlock}`);
+  const searchConfig = { fromBlock, toBlock, maxBlockLookBack: maxBlockRange };
+
+  const tStart = performance.now();
+  const filter = getEventFilter(spokePool, eventName, filterArgs[eventName]);
+  const events = await paginatedEventQuery(spokePool, filter, searchConfig);
+  const tStop = performance.now();
+  logger.debug({
+    at: "SpokePoolIndexer::listen",
```
Comment (pxrl) [inline src/libexec/util/evm/util.ts:64]:
```suggestion
    at: "scrapeEvents",
```

File: src/libexec/util/evm/util.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ */
+export async function scrapeEvents(
+  spokePool: Contract,
+  eventName: string,
+  opts: ScraperOpts & { toBlock: number },
+  logger: winston.Logger
+): Promise<Log[]> {
+  const { lookback, deploymentBlock, filterArgs, maxBlockRange, toBlock } = opts;
+  const { provider } = spokePool;
+  const { chainId } = await provider.getNetwork();
```
Comment (pxrl) [inline src/libexec/util/evm/util.ts:52]:
```suggestion
  const { chainId } = await spokePool.provider.getNetwork();
```

**Review by bmzig — APPROVED** [review]

**Review by nicholaspai — APPROVED** [review]
I like the evm/svm distinction

File: src/libexec/types.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export { Log } from "../interfaces";
+export { SpokePoolClientMessage } from "../clients";
+
+export type ScraperOpts = {
+  lookback?: number; // Event lookback (in seconds).
+  deploymentBlock: number; // SpokePool deployment block
```
[the remark this answers is not in the document]
Comment (pxrl) [reply src/libexec/types.ts:6]: I'm not sure; I've moved them there optimistically but I fully expect we'll need some adjustment once SVM parts start to get dropped in here.

**Pull request merged**